test/functional/test_framework/muhash.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2020 Pieter Wuille
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Native Python MuHash3072 implementation."""
+
+from .util import (
+    assert_equal,
+    modinv,
+)
+
+import hashlib
```

Comment:
Normally, standard library modules are imported before local project modules: https://www.python.org/dev/peps/pep-0008/#imports

Reply:
Yes, further on this, a couple of Python linters I've been finding useful are pycodestyle (`pip install pycodestyle`) and black (`pip install black`), not necessarily to follow blindly but they do provide helpful suggestions.